src/Math-Complex/PMComplex.class.st

```diff
@@ -694,14 +713,14 @@ PMComplex >> sinh [
 
 { #category : #'mathematical functions' }
 PMComplex >> sqrt [
+
 	"Return the square root of the receiver with a positive imaginary part."
 
 	| u v |
-	(imaginary = 0 and: [real >= 0])
-		ifTrue:	[^self class real: real sqrt imaginary: 0].
+	self isReal ifTrue: [ ^ self class real: real sqrt imaginary: 0 ].
```

Comment:
I think you extract ``(imaginary = 0 and: [real >= 0])`` a bit too fast, this is not replaced by isReal but by isReal and positive.

Reply:
How about `z isZero or: z isReal` ?

Reply:
How about `z isZero or: z isReal`?

Reply:
I don't know if this is necessary. Maybe keep it like it was before.

Reply:
Sure. Perhaps we need more tests before refactoring.

Reply:
I have inlined the method for us.